pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -32,14 +36,32 @@ func ConvertEnvs(vars []common.DockerimageEnv) []corev1.EnvVar {
 	return kVars
 }
 
+// ConvertPorts converts endpoint variables from the devfile structure to kubernetes ContainerPort
+func ConvertPorts(endpoints []common.DockerimageEndpoint) []corev1.ContainerPort {
+	containerPorts := []corev1.ContainerPort{}
+	for _, endpoint := range endpoints {
+		name := strings.TrimSpace(util.GetDNS1123Name(strings.ToLower(*endpoint.Name)))
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          name,
```

Comment:
does this need to be trimmed to below 63 char? I had a similar issue for "vol name", I later found out not only Kubernetes resources but other names are also capped at 63. Can you pls confirm this?

Reply:
Thanks for pointing this out. Tested that the max length for port name is 15 characters. Push a new commit to trim it down to 15 characters.